### PR TITLE
APP-1008 bootstrap retry

### PR DESCRIPTION
**Pull request by ecarrenho-daitan**
- Adding logic to handle retry when bot is unauthorized.
- Fixing trace id to truncate concatenation of a maximum of 3 thread ids.

### CODE REVIEW

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java [hunk 1]
```diff
@@ -183,13 +181,7 @@ private void authenticate(String integrationUser) {
...
       authenticationProxy.authenticate(integrationUser);
     } catch (RemoteApiException e) {
-      exceptionHandler.handleAuthenticationApiException(integrationUser, e);
-    } catch (ConnectivityException e) {
-      // Needed to rethrow this specific exception, otherwise it will endup hidden under the
-      // catch bellow.
-      throw e;
-    } catch (Exception e) {
-      exceptionHandler.handleAuthException(e);
+      throw new RetryLifecycleException("Unexpected error when authenticating", e);
```
Comment (rsanchez-daitan) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java]: I believe you should catch Exception and throw a UnexpectedAuthException exception.

This way you can define a well-defined message and in the future we can define a solution.

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java [hunk 2]
```diff
@@ -389,14 +381,8 @@ private void postMessage(IntegrationInstance instance, String integrationUser,
...
-    List<Message> response = new ArrayList<>();
-
-    try {
-      // Post a message
-      response = bridge.sendMessage(instance, integrationUser, streams, message);
-    } catch (ProcessingException e) {
-      exceptionHandler.handleAuthException(e);
-    }
+    // Post a message
+    List<Message> response = bridge.sendMessage(instance, integrationUser, streams, message);
```
Comment (rsanchez-daitan) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java]: Same comment above.

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java (hunk 1 above)
Comment (ecarrenho-daitan) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java]: Done.

**Review by robson-sanchez — APPROVED** [review]
LGTM

**Pull request merged**